### DISCUSSION TITLE
Curso FullStack - Requisição de Categorias

### DISCUSSION THREAD

**Discussion by mamsgruppe** [balta-io/forum#572] [Tira Dúvidas] [answered]
É possível criar um único arquivo chamado CategoriesResquest e dentro dele ter o:

1- CreateCategoryRequest;
2- DeleteCategoryRequest;
3- UpdateCategoryRequest:

Ou indo um pouco mais além, poderia ter um pasta Pai chamada DTO ao invés de Request/Response e dentro dela ter apenas um arquivo chamado de CategoriesDTO com todos os Dtos Request/Response;

Qual a recomendação citando os pros e contras de cada item acima? Por favor não respondam "depende do caso ou projeto"...rsrsrs...listando os pros e contras ficará mais fácil a avaliação/decisão para tomada de decisão.

![Screenshot 2024-06-06 at 20 52 26](https://github.com/balta-io/forum/assets/155583120/7535e474-b77d-40bf-99d6-040ae4f87e50)

**Comment by malighieri** [comment, marked as answer]
Minha resposta não vai ser diretamente depende, mas vai ser quase a mesma coisa. rs
Quando você está estruturando a arquitetura de um software você pode adotar algum padrão de organização/arquitetura de acordo com como você enxerga o contexto que o software atende. Não chega a ter um certo e errado mas existem boas práticas, recomendações e padrões mais usados. Por exemplo, você pode adotar Arquitetura Limpa, Arquitetura Vertical, Arquitetura Hexagonal, etc... Cada uma aborda a organização do software focando em um aspecto. É legal saber o que cada uma propõe e entender qual melhor representa sua visão (ou a do time) sobre o software

**Comment by andrebaltieri** [comment]
@mamsgruppe pode sim, sem problemas, a ideia maior é que você saiba o que está fazendo e para o que cada coisa serve... os nomes e onde ficam, é um detalhe.

Cada implementação vai ter seus prós e contras, e não tem um jeito 100% correto.

Minha dica: TESTE!!!! 
Use e abuse do que aprendeu, varie em cima disso, crie diferentes implementações... experimente...
O curso é só a minha visão, fazendo isso que falei, você vai desenvolver a sua 🚀
